lib/global-styles.php

```diff
@@ -235,6 +243,9 @@ function_exists( 'gutenberg_is_edit_site_page' ) &&
 	return $settings;
 }
 
-add_action( 'init', array( 'WP_Theme_JSON_Resolver', 'register_user_custom_post_type' ) );
+if ( gutenberg_experimental_global_styles_has_theme_json_support() ) {
```

Comment:
I think the function gutenberg_experimental_global_styles_has_theme_json_support  must be called inside actions. Right now with this change, I'm getting some notices:
```

  | <br />
-- | --
  | <b>Notice</b>:  Use of undefined constant STYLESHEETPATH - assumed 'STYLESHEETPATH' in <b>/Users/pc/dev/core/wordpress-develop/build/wp-includes/template.php</b> on line <b>663</b><br />
  | <br />
  | <b>Notice</b>:  Use of undefined constant TEMPLATEPATH - assumed 'TEMPLATEPATH' in <b>/Users/pc/dev/core/wordpress-develop/build/wp-includes/template.php</b> on line <b>666</b><br />
  | <br />
  | <b>Warning</b>:  Cannot modify header information - headers already sent by (output started at /Users/pc/dev/core/wordpress-develop/build/wp-includes/template.php:663) in <b>/Users/pc/dev/core/wordpress-develop/build/wp-includes/functions.php</b> on line <b>6360</b><br />
  | <br />
  | <b>Warning</b>:  Cannot modify header information - headers already sent by (output started at /Users/pc/dev/core/wordpress-develop/build/wp-includes/template.php:663) in <b>/Users/pc/dev/core/wordpress-develop/build/wp-admin/includes/misc.php</b> on line <b>1310</b><br />
  | <br />
  | <b>Warning</b>:  Cannot modify header information - headers already sent by (output started at /Users/pc/dev/core/wordpress-develop/build/wp-includes/template.php:663) in <b>/Users/pc/dev/core/wordpress-develop/build/wp-admin/admin-header.php</b> on line <b>9</b><br />

```

I think it happens because some constants required are not initialized during this code execution.